src/transaction/log_recovery_analysis.cpp

```diff
@@ -554,6 +557,45 @@ log_rv_analysis_undo_redo (THREAD_ENTRY *thread_p, int tran_id, LOG_LSA *log_lsa
   if (is_mvcc)
     {
       tdes->last_mvcc_lsa = *log_lsa;
+
+      // assign transaction mvccid from log record to transaction descriptor
+      assert (log_page_p != nullptr);
+
+      // move read pointer past the log header which is actually read upper in the stack
+      LOG_READ_ADD_ALIGN (thread_p, sizeof (LOG_RECORD_HEADER), log_lsa, log_page_p);
+      switch (log_type)
+	{
+	case LOG_MVCC_UNDOREDO_DATA:
+	case LOG_MVCC_DIFF_UNDOREDO_DATA:
+	{
+	  // align to read the specific record info
+	  LOG_READ_ADVANCE_WHEN_DOESNT_FIT (thread_p, sizeof (LOG_REC_MVCC_UNDOREDO), log_lsa, log_page_p);
+	  const LOG_REC_MVCC_UNDOREDO *const log_rec
+	    = (const LOG_REC_MVCC_UNDOREDO *) ((char *)log_page_p->area + log_lsa->offset);
+	  tdes->mvccinfo.id = log_rec->mvccid;
```

Comment:
Does a code exist (and is executed after all log analysis is complete) where each TDES is checked if has a valid mvccid ?

Reply:
There is no such check.
Is it needed?
The way I understood it is that the vast majority of transactions do have an mvccid, but I also encountered transactions that will not have an mvccid (eg: I think the checkpoint transaction but I'm not sure).
There was an assert that the mvccid must be valid in the `log_rv_analysis_complete` function but it was raised when such a transaction commited.